helm-chart/templates/web.yaml

```diff
@@ -0,0 +1,216 @@
+            protocol: TCP
+        resources:
+          requests:
+            memory: {{ .Values.web.resources.web.request.memory }}
+            cpu: {{ .Values.web.resources.web.request.cpu }}
+          limits:
+            memory: {{ .Values.web.resources.web.limit.memory }}
+            cpu: {{ .Values.web.resources.web.limit.cpu }}
+        # turn off health checks for content sandboxes
+        {{ if ne .Values.contentSandbox "true" }}
+        # set failure thresholds
+        startupProbe:
+          httpGet:
+            path: /healthz-rollout
+            port: 80
+          failureThreshold: 40 # 40 attempts x 15 seconds = 10 minutes for the pod to successfully start
+          periodSeconds: 15
+          timeoutSeconds: 10
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 80
+          # initialDelaySeconds: 120
+          periodSeconds: 60
+          timeoutSeconds: 60
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: 80
+          # initialDelaySeconds: 120
+          periodSeconds: 60
+          timeoutSeconds: 60
+        {{ end }}
+        volumeMounts:
+        # https://stackoverflow.com/questions/53491603/kubernetes-volumemounts-a-file-not-a-directory
+        - mountPath: /app/sefaria/local_settings.py
+          name: local-settings
+          subPath: local_settings.py
+          readOnly: true
+        - mountPath: /varnish-secret
+          name: varnish-secret
+          readOnly: true
+        - mountPath: /school-lookup-data
+          name: school-lookup-data
+          readOnly: true
+        - mountPath: /client-secret
+          name: client-secret
+          readOnly: true
+        - mountPath: /google-cloud-secret
+          name: backup-manager-secret
+          readOnly: true
+        - mountPath: /app/logging-secret.json
+          name: logging-secret
+          subPath: logging-secret.json
+          readOnly: true
+        - name: gunicorn-config
+          mountPath: /app/gunicorn.conf.py
+          subPath: gunicorn.conf.py
+          readOnly: true
+      volumes:
+        - name:  local-settings
+          secret:
+            secretName: local-settings-{{ .Values.deployEnv }}
+        - name: client-secret
+          secret:
+            secretName: google-client-secret-{{ .Values.deployEnv }}
+        - name: backup-manager-secret  # used to access google cloud
+          secret:
+            secretName: backup-manager-secret-{{ .Values.deployEnv }}
+        - name: logging-secret
+          secret:
+            secretName: logging-secret-{{ .Values.deployEnv }}
+        - name: varnish-secret
+          secret:
+            secretName: varnish-secret-{{ .Values.deployEnv }}
+        - name: school-lookup-data
+          secret:
+            secretName: school-lookup-data-{{ .Values.deployEnv }}
+        - name: gunicorn-config
+          configMap:
+            Name: gunicorn-config-{{ .Values.deployEnv }}
```

Comment:
All secret references should contain a toggle to either a reference to an existing secret, or the envNamed secret generated if data is supplied.